### DISCUSSION TITLE
XAML support (Avalonia, WPF, UWP)

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [mallowigi/color-highlighter#74] [Ideas]
Having the support for XAML frameworks would be awesome and of use.

As currently there's no highlighter for Avalonia.

Code example
```xml
<Border  Background="#A1202020" <- Highlight this color
         BorderBrush="#707070" <- Highlight this color
         BorderThickness="1">
            
</Border>
```

**Comment by mallowigi** [comment]
I've received a similar request in the past, but I couldn't find a way to create XAML applications with Rider. Best I could do was either Xamarin or ASP.NET

If you could enlighten me on how to create such things, it would be useful.

**Reply by PerfectlyFineCode** [reply]
Avalonia is a cross-platform UI framework for .NET which is based off of XAML, very similar to WPF. It also supports JetBrains Rider. That's where the example is from.

To set it up in Rider, check out the link posted below.
https://docs.avaloniaui.net/docs/getting-started/ide-support/jetbrains-rider-setup

Dotnet CLI can also help you create other projects, such as new project from the WPF template.

**Reply by mallowigi** [reply]
Thanks. I'll try to do it whenever I have the time, but its not going to be anytime soon. You are free to try by yourself though :)
